src/Access/AccessControlHandler.php

```diff
@@ -82,18 +59,34 @@ public function hasPasswordChangeAccess(AccountInterface $current_user): AccessR
   }
 
   /**
-   * Checks whether the user can create any more policies or not.
+   * Check whether the given role can have one more given password constraint.
    *
-   * @param \Drupal\Core\Session\AccountInterface $current_user
-   *   The current user.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $current_route_match
+   *   The current route match service.
    *
    * @return \Drupal\Core\Access\AccessResultInterface
    *   Access result.
    */
-  public function canCreatePolicy(AccountInterface $current_user): AccessResultInterface {
-    $policies = $this->policyEntityStorage->getQuery()->count()->execute();
-    $roles = $this->roleStorage->getQuery()->condition('id', Role::ANONYMOUS_ID, '<>')->count()->execute();
-    return AccessResult::allowedIf($policies < $roles)->setCacheMaxAge(0);
+  public function hasConstraintAddAccess(RouteMatchInterface $current_route_match): AccessResultInterface {
+    $allowed = FALSE;
+    $constraint_type = $current_route_match->getParameter('password_constraint');
+    $policy = PasswordPolicy::createFromRole($current_route_match->getParameter('user_role'));
+    // User can add the given constraint to the given role with a policy when:
+    // - the constraint type is not unique,
+    // - OR the policy doesn't have this type of a constraint yet.
+    if ($policy) {
+      $allowed = TRUE;
+      if ($constraint_type['unique']) {
+        /** @var \Drupal\password_enhancements\Plugin\PasswordConstraint\PasswordConstraintBase $policy_constraint */
+        foreach ($policy->getConstraints() as $policy_constraint) {
+          if ($policy_constraint->getPluginId() == $constraint_type['id']) {
+            $allowed = FALSE;
+            break;
+          }
+        }
+      }
+    }
+    return AccessResult::allowedIf($allowed);
```

Comment:
Are you sure that it is ok if `$allowed` is `false`, then the return value is a neutral access result?

Reply:
Yes, because that prevented me from reaching the form – IOW, I'm getting a 403 in that case as expected.